src/middlewared/middlewared/plugins/account.py

```diff
@@ -1555,6 +1573,22 @@ async def query(self, filters, options):
             else:
                 entry.update({'nt_name': None, 'sid': None})
 
+            roles = []
```

Comment:
Can we avoid duplicating this logic by using `privileges_for_groups`? Possibly refactoring some logic out there for performance reasons.

Reply:
We can perhaps have hundreds of local users (I think we had a user with over 500). I'd rather not do N calls to privilege endpoint.

Reply:
Okay. I see how I can do this without performance hit.